mods/explorer-font-changer.wh.cpp

```diff
@@ -37,64 +56,317 @@ Simply change the font name to "None", or turn off the plugin.
 # https://github.com/ramensoftware/windhawk/wiki/Creating-a-new-mod#settings
 - font:
   - name: "None"
+    $name: "Font Face Name"
+  - glow: false
+    $name: "Glow text"
+  - r: 255
+    $name: "Glow: Red (0-255)"
+  - g: 255
+    $name: "Glow: Green (0-255)"
+  - b: 255
+    $name: "Glow: Blue (0-255)"
+  - a: 255
+    $name: "Glow: Alpha (0-255)"
+  - intensity: 4
+    $name: "Glow: Intensity (1-..)"
   $name: Custom font
   $description: This font will be used for all font creation in Windows.
 */
 // ==/WindhawkModSettings==
 
-// The source code of the mod starts here. This sample was inspired by the great
-// article of Kyle Halladay, X64 Function Hooking by Example:
-// http://kylehalladay.com/blog/2020/11/13/Hooking-By-Example.html
-// If you're new to terms such as code injection and function hooking, the
-// article is great to get started.
-
 #include <string>
 #include <windhawk_utils.h>
+#include <uxtheme.h>
 
 using namespace std::string_view_literals;
 using namespace WindhawkUtils;
 
-using create_font_indirectw_hook_t = decltype(&CreateFontIndirectW);
-create_font_indirectw_hook_t create_font_indirectw_original = nullptr;
+namespace util {
+    // Class to declare transparent RAII types.
+    template <class T, auto Fn>
+    class raii {
+    public:
+        raii() = delete;
+        raii(T handle) : m_handle(handle) {}
+        ~raii() {
+            Fn(m_handle);
+        }
 
-HFONT WINAPI create_font_indirectw_hook(const LOGFONTW* _font) {
-    // Cast the const away for operations. Could be unsafe.
-    auto font = const_cast<LOGFONTW*>(_font);
+        // we don't need operator= or raii(raii&&) yet...
 
-    auto set_font_name = StringSetting::make(L"font.name");
-    auto font_name = std::wstring_view(set_font_name.get());
+        auto& get() {
+            return m_handle;
+        }
+        
+        auto get() const {
+            return m_handle;
+        }
+
+    private:
+        T m_handle;
+    };
+
+    // RAII `HFONT`.
+    using unique_hfont_t = raii<HFONT, DeleteObject>;
```

Comment:
FYI you can also use [WIL](https://github.com/microsoft/wil) which is included with Windhawk. It has types such as [`unique_hfont`](https://github.com/search?q=repo%3Amicrosoft%2Fwil+unique_hfont&type=code). Currently there's [one known issue](https://github.com/ramensoftware/windhawk/issues/553) with it but there's also a workaround.

Reply:
Seems like a huge headache for now, honestly